tests/test_sync_followers_command.py

```diff
@@ -0,0 +1,151 @@
+
+        call_command("sync_followers", "FormalEmailProcess", verbosity=2, dry_run=True)
+
+        # Neither of the follower apps have been synced.
+        self.assertEqual(formal_emails.recorder.max_tracking_id("BankAccounts"), 0)
+        self.assertEqual(informal_emails.recorder.max_tracking_id("BankAccounts"), 0)
+
+    def test_sync_all_apps(self) -> None:
+        leader_notification_id = self._create_leader_events()
+        formal_emails = self.runner.get(FormalEmailProcess)
+        informal_emails = self.runner.get(InformalEmailProcess)
+
+        # Get the system running.
+        self.runner.start()
+
+        # Both follower apps are unaware of the leader events.
+        self.assertEqual(formal_emails.recorder.max_tracking_id("BankAccounts"), 0)
+        self.assertEqual(informal_emails.recorder.max_tracking_id("BankAccounts"), 0)
+
+        call_command("sync_followers", verbosity=2)
+
+        # Both follower apps have been synced.
+        self.assertEqual(
+            formal_emails.recorder.max_tracking_id("BankAccounts"),
+            leader_notification_id,
+        )
+        self.assertEqual(
+            informal_emails.recorder.max_tracking_id("BankAccounts"),
+            leader_notification_id,
+        )
+
+    def test_dry_run_sync_all_apps(self) -> None:
+        self._create_leader_events()
+        formal_emails = self.runner.get(FormalEmailProcess)
+        informal_emails = self.runner.get(InformalEmailProcess)
+
+        # Get the system running.
+        self.runner.start()
+
+        # Both follower apps are unaware of the leader events.
+        self.assertEqual(formal_emails.recorder.max_tracking_id("BankAccounts"), 0)
+        self.assertEqual(informal_emails.recorder.max_tracking_id("BankAccounts"), 0)
+
+        call_command("sync_followers", verbosity=2, dry_run=True)
+
+        # Neither of the follower apps have been synced.
+        self.assertEqual(formal_emails.recorder.max_tracking_id("BankAccounts"), 0)
+        self.assertEqual(informal_emails.recorder.max_tracking_id("BankAccounts"), 0)
+
+
+class TestWithSQLiteFileDb(TestSyncCommand):
+    django_db_alias = "sqlite_filedb"
+    databases = {"default", "sqlite_filedb"}
+
+    @pytest.mark.xfail(reason="Transaction not rolled back?")
+    def test_dry_run_sync_one_app(self) -> None:
+        super().test_dry_run_sync_one_app()
+
+    @pytest.mark.xfail(reason="Transaction not rolled back?")
+    def test_dry_run_sync_all_apps(self) -> None:
+        super().test_dry_run_sync_all_apps()
+
+
+class TestWithPostgres(TestSyncCommand):
+    django_db_alias = "postgres"
+    databases = {"default", "postgres"}
+
+    @pytest.mark.xfail(reason="Transaction not rolled back?")
+    def test_dry_run_sync_one_app(self) -> None:
+        super().test_dry_run_sync_one_app()
+
+    @pytest.mark.xfail(reason="Transaction not rolled back?")
+    def test_dry_run_sync_all_apps(self) -> None:
+        super().test_dry_run_sync_all_apps()
```

Comment:
I'm willing to work on these rollbacks if need be. Not sure why this doesn't work, or what the right answer is just now. But I'm sure we can figure this out together, if you can't figure it out on your own.

Reply:
Great, that's a nice fix!